src/cordova/restore-util.js

```diff
@@ -341,19 +340,29 @@ function installPluginsFromConfigXML (args) {
             fs.writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, null, 2), 'utf8');
         }
     }
-    // Write config.xml (only if plugins exist in package.json).
+    // Write to config.xml (only if it is different from package.json in content)
     comboPluginIdArray.forEach(function (plugID) {
+        var configXMLPlugin = cfg.getPlugin(plugID);
         if (pluginIdConfig.indexOf(plugID) < 0) {
             pluginIdConfig.push(plugID);
-        }
-        cfg.removePlugin(plugID);
-        if (mergedPluginSpecs[plugID]) {
+            if (mergedPluginSpecs[plugID]) {
+                cfg.removePlugin(plugID);
+                cfg.addPlugin({name: plugID, spec: mergedPluginSpecs[plugID]}, comboObject[plugID]);
+                modifiedConfigXML = true;
+            } else {
+                cfg.removePlugin(plugID);
+                cfg.addPlugin({name: plugID}, comboObject[plugID]);
+                modifiedConfigXML = true;
+            }
+
+        // Write only if the plugin variables or specs are different from pkgJson
+        } else if (((pluginIdConfig.indexOf(plugID) > 0) && (mergedPluginSpecs[plugID]) &&
+            ((configXMLPlugin.variables !== comboObject[plugID]))) ||
+            ((mergedPluginSpecs[plugID] !== configXMLPlugin.spec) ||
```

Comment:
so to confirm, either `((pluginIdConfig.indexOf(plugID) > 0) && (mergedPluginSpecs[plugID]) &&(configXMLPlugin.variables !== comboObject[plugID]))` or `((mergedPluginSpecs[plugID] !== configXMLPlugin.spec) || (configXMLPlugin.variables !== comboObject[plugID]))` need to be true